### PR TITLE
Automatic checks

### PR DESCRIPTION
**Pull request by ru-fu**

### CODE REVIEW

**Comment by ru-fu** [comment]
@MathieuBordere @cole-miller 
Some changes to the docs to satisfy the spell checker. ;-)

I think most changes are straight-forward, just two things we should align on:

- "Dqlite" or "dqlite"? The website (and the majority of the docs) use "Dqlite", the GitHub repo uses "dqlite".
   I suggest to use "Dqlite" and "the `dqlite` library".
- British English or American English? The default for Canonical's documentation is British English. LXD is an exception, since it isn't a Canonical product.

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  spelling:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```
Comment (MathieuBordere) [inline .github/workflows/spelling.yml:11]: I think Github will complain about the version of checkout, latest version is `v3`.

File: .github/workflows/woke.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - push
+  - pull_request
+
+jobs:
+  woke:
+    name: woke
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (MathieuBordere) [inline .github/workflows/woke.yml:13]: should use v3 here too

File: consistency-model.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
-More precisely, Dqlite transactions are **serializable**, however their are **not linearizable**. They are serializable because Dqlite allows at most one writer at a time and readers always observe a consistent (i.e. committed) snapshot of the database, therefore transactions can't "interleave". They are not linearizable because there's no guarantee that a read transaction will always observe a snapshot of the most recently committed version of the database that existed when the read transaction started. In other words, **stale reads** are allowed.
+More precisely, Dqlite transactions can be *serialised*, however they *cannot be linearised*. They are serialisable because Dqlite allows at most one writer at a time and readers always observe a consistent (i.e. committed) snapshot of the database, therefore transactions can't "interleave". They are not linearisable because there's no guarantee that a read transaction will always observe a snapshot of the most recently committed version of the database that existed when the read transaction started. In other words, *stale reads* are allowed.
```
Comment (MathieuBordere) [inline consistency-model.md:9]: serializable and linearizable (American English) are database lingo, serialised and linearised don't feel right imo.

File: protocol.md [hunk 4]
```diff
@@ -2,18 +2,18 @@
...
+* *protocol version* : A positive number stored in a word using little endian representation.
+* *message* : A sequence of bytes sent either by the client to the server or by the server to the client. It consists of a header and a body. The header consists of a single word with the following layout:
   * byte 0 to 3: Size of the message body, expressed in words and stored using little endian representation. For example a value of [2 1 0 0] means that the message body consists of 258 bytes.
-  * byte 4: Type code identifiying the schema of the message body.
+  * byte 4: Type code identifying the schema of the message body.
   * byte 5: Message schema version. This version is bumped for a particular message type when the expected format of the body for that message type changes. Unless otherwise noted, the message formats described in this document are for schema version 0.
   * byte 6 to 7: Currently unused.
 
-The message body is a sequence of fields as described by the associated schema. Message types and their schemas are listed below.
+The message body is a sequence of fields as described by the associated schema. Message types and their schemata are listed below.
```
Comment (MathieuBordere) [inline protocol.md:16]: schemata feels a bit archaic?

File: protocol.md [hunk 5]
```diff
@@ -27,323 +27,322 @@ After the setup, communication between client and server happens by message exch
...
-| --- | --- |
-|uint64|New weight|
+| Type     | Value      |
+| ---      | ---        |
+| `uint64` | New weight |
 
 ## Server messages
 
-The server can send to the client messages with the following type codes and associated schemas:
+The server can send to the client messages with the following type codes and associated schemata:
```
Comment (MathieuBordere) [inline protocol.md:260]: same remark as before

File: replication.md [hunk 6]
```diff
@@ -18,25 +18,25 @@ Dqlite configures SQLite to use a custom VFS implementation which stores files i
...
+In this mode SQLite uses two files to store data: one is the *main database file*, and the other is the *WAL file*. When an SQLite transaction that modifies the database wants to commit those changes, it does not write the database pages it has modified directly to the database file, but it rather appends them to the WAL file. The last page of a write transaction appended to the WAL will contain a *commit marker* that marks the end of the transaction. Once the WAL file grows too big, a checkpoint must be created, and the latest version of each page of each committed transaction it contains gets copied back to the main database file.
```
Comment (MathieuBordere) [inline replication.md:27]: suggest to change "a checkpoint must be created," to "a checkpoint operation must be performed,"

**Review by MathieuBordere — COMMENTED** [review]
Thanks! Some small comments.

File: consistency-model.md (hunk 3 above)
Comment (ru-fu) [reply consistency-model.md:9]: Seems like BE allows "serialize" as well, and "linearise/linearize" isn't really a word anyway, so I'm fine with going for the "z" variant for those two.

File: protocol.md (hunk 4 above)
Comment (ru-fu) [reply protocol.md:16]: OK, I guess aspell is a bit archaic - the current dictionaries actually allow both. :)
Changing back.

**Comment by MathieuBordere** [comment]
LGTM!

**Pull request merged**